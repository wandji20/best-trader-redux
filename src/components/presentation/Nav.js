import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Option from './Option';
import { filterByCurrencyAction, filterByMarketAction } from '../../redux/action';

const Nav = (props) => {
  const { markets, filterByCurrency, filterByMarket } = props;

  const getMarketPickers = (markets) => markets.map((market) => market.ticker);

  const tickers = getMarketPickers(markets);
  const getCurrency = (tickers) => tickers.map((ticker) => ticker.split('/'));

  const allCurrencies = getCurrency(tickers).flat();
  const currencies = allCurrencies.filter(
    (currency, index) => allCurrencies.indexOf(currency) === index,
  );

  const handleCurrencyChange = (e) => {
    filterByCurrency(e.target.value);
  };

  const handleMarketChange = (e) => {
    filterByMarket(e.target.value);
  };

  return (
    <header className="container remove-padding bg-dark text-white">
      <nav className="container-fluid d-flex justify-content-between py-3">
        <div className="d-flex align-items-center justify-content-between w-sm-100 w-75">
          <h5 className="d-none d-md-flex">
            <a href="/" className="text-info">
              <span className="">
                Best Trader
              </span>
              <span className="d-inline-block fs-6 fw-1">
                TM
              </span>
            </a>
          </h5>
          <div className="d-flex justify-content-center align-items-center">
            <div className="input-group">
              <input
                onChange={handleMarketChange}
                type="text"
                className="form-control"
                placeholder="market"
              />
            </div>

            <select
              onChange={handleCurrencyChange}
              className="form-select nav-item mx-3"
            >
              <option key="All">All</option>
              {
                currencies.sort()
                  .map((currency) => (
                    <Option
                      key={currency}
                      ticker={currency}
                    />
                  ))
              }
            </select>
          </div>
        </div>
      </nav>
    </header>
  );
};

const mapStateToProps = (state) => ({
  markets: state.forexReducer.markets,
});

const mapDispatchToProps = (dispatch) => ({
  filterByCurrency: (data) => {
    dispatch(filterByCurrencyAction(data));
  },
  filterByMarket: (data) => {
    dispatch(filterByMarketAction(data));
  },
});

Nav.propTypes = {
  markets: PropTypes.arrayOf(PropTypes.object).isRequired,
  filterByCurrency: PropTypes.func.isRequired,
  filterByMarket: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
