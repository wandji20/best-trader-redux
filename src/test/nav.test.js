import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Nav from '../components/presentation/Nav';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Nav /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
