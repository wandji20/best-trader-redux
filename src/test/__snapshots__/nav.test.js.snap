// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="container remove-padding bg-dark text-white"
>
  <nav
    className="container-fluid d-flex justify-content-between py-3"
  >
    <div
      className="d-flex align-items-center justify-content-between w-sm-100 w-75"
    >
      <h5
        className="d-none d-md-flex"
      >
        <a
          className="text-info"
          href="/"
        >
          <span
            className=""
          >
            Best Trader
          </span>
          <span
            className="d-inline-block fs-6 fw-1"
          >
            TM
          </span>
        </a>
      </h5>
      <div
        className="d-flex justify-content-center align-items-center"
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="market"
            type="text"
          />
        </div>
        <select
          className="form-select nav-item mx-3"
          onChange={[Function]}
        >
          <option>
             
          </option>
          <option>
            All
          </option>
        </select>
      </div>
    </div>
  </nav>
</header>
`;
